Feature/suitep - Colors

daniel-samson opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here unless your commit contains the issue number -->
Updated the sass to support multiple color schemes in SuiteP.

I have split the colors in to a subfolders dawn, day, dusk, night. each folder has
- style.css - includes the styles for that color scheme.
- color-palette - includes a list of colors
- variables.css - includes the rules for the colors which reference the color-palette.

Notice that we can now use search and replace to  colors in the color pallete. However if you need to change the which elements use a color, you can do that in the variables.scss

The actual style sheets are in the bootstrap and suitep-base folders. These reference variable in the variables.scss.


## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
- Gives user the choice of there favorite color scheme.
- Allows the community to create their own colors

## How To Test This
<!--- Please describe in detail how to test your changes. -->
- Go to the User profile
- Click on advanced
- Change style

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
<!--- Go over all the following points and check all the boxes that apply. --->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! --->
- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [x] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

<!--- Your pull request will be tested via Travis CI to automatically indicate that your changes do not prevent compilation. --->

<!--- If it reports back that there are problems, you can log into the Travis system and check the log report for your pull request to see what the problem was. --->

Dillon-Brown commented:
@daniel-samson 

Issue 1: Required fields * is the same colour on every theme.

Issue 2: Email compose view text field is unstyled.
![issue2](https://user-images.githubusercontent.com/26431166/33833775-a5dce34a-de78-11e7-8333-80307c0fdf7a.png)

Issue 3: View summary style is broken.
![issue3](https://user-images.githubusercontent.com/26431166/33833787-b918b722-de78-11e7-9d1c-f0f3f08a5edf.png)

Issue 4: Small area in reports is unstyled
![issue4](https://user-images.githubusercontent.com/26431166/33833809-cb873e60-de78-11e7-8a5b-e6751f127cf3.png)

Issue 5: Small area in campaigns is unstyled
![issue5](https://user-images.githubusercontent.com/26431166/33833817-d1557622-de78-11e7-96f5-56573e837bae.png)

daniel-samson commented:
@Dillon-Brown 

Issue 1: 
I have added a new variable into the variables.scss. Currently it will be red in all themes, However it is easy to change. There is not a specified color for this in the design.

Issue 2:
I have added a new skin to remove the colors and added a new sass file to control the colors

Issue 3:
This was caused by the tpl in the activities module. It did not use the SugarThemeRegistry to get the css info. As a result it tried to load the style.css from the old path. This has now been corrected.

Issue 4:

I have corrected the color issue here.

Issue 5:
I have corrected the color issue here.

gymad commented on config_override.php:
```diff
@@ -0,0 +1,5 @@
+<?php
```
this file shouldn't in PR

gymad commented on include/SugarTheme/SugarThemeRegistry.php:
```diff
@@ -0,0 +1,390 @@
+<?php
+
```
missing licence header

gymad commented on include/SugarTheme/SugarThemeRegistry.php:
```diff
@@ -0,0 +1,390 @@
+    /**
+     * Disable the constructor since this will be a singleton
+     */
+    private function __construct() {}
```
empty constructor?

gymad commented on themes/SuiteP/css/bootstrap/theme.css.map:
```diff
@@ -0,0 +1,7 @@
+{
```
map files should be excluded from PR

gymad commented on themes/SuiteP/css/Dawn/variables.css:
```diff
@@ -0,0 +1,7 @@
+/*!
```
why this empty css for?

daniel-samson replied:
removed

gymad commented on themes/SuiteP/css/studio.css:
```diff
@@ -1,201 +1,67 @@
+body:before {
+  white-space: pre;
+  font-family: monospace;
+  content: "Error: Undefined variable: \"$page-font-size\".\A         on line 13 of /var/www/html/sdk/instances/codeception/themes/SuiteP/css/studio.scss\A \A 8:   width: 32px;\A 9:   height: 32px;\A 10: }\A 11: \A 12: #exportcustom h3 {\A 13:   font-size: $page-font-size + 1px;\A 14:   font-weight: bold;\A 15:   padding-left: 0px;\A 16:   margin: 3px;\A 17: }\A 18: "; }
```
error?

daniel-samson replied:
reverted file

gymad commented on themes/default/css/content.min.css:
```diff
@@ -0,0 +1 @@
+/* use to remove skin from tinemce*/
```
one comment line css?

daniel-samson replied:
We need to clear the tine mce skin so that the sass can define the style instead...
